### PR TITLE
build: make lda optional with compile feature

### PR DESCRIPTION
**Pull request by jackgerrits**

### CODE REVIEW

File: cmake/VowpalWabbitFeatures.cmake [hunk 1]
```diff
@@ -8,17 +8,12 @@
...
-#   CSV: Enable csv parser
-#   LAS_SIMD: Enable large action space with explicit simd (only work with linux for now)
 
-# set(VW_ALL_FEATURES "LAS_SIMD;FLATBUFFERS;CSV")
-set(VW_ALL_FEATURES "CSV;FLATBUFFERS")
+set(VW_ALL_FEATURES "CSV;FLATBUFFERS;LDA")
 
 option(VW_FEAT_FLATBUFFERS "Enable flatbuffers support" OFF)
 option(VW_FEAT_CSV "Enable csv parser" OFF)
+option(VW_FEAT_LDA "Enable lda reduction" ON)
```
Comment (bassmang) [inline cmake/VowpalWabbitFeatures.cmake:16]: Is there a reason to not set this OFF by default?

File: ext_libs/README [hunk 2]
```diff
@@ -0,0 +1,9 @@
+boost_math - optional
```
Comment (lalo) [inline ext_libs/README:1]: thoughts?

File: cmake/VowpalWabbitFeatures.cmake (hunk 1 above)
Comment (jackgerrits) [reply cmake/VowpalWabbitFeatures.cmake:16]: I wanted to maintain current behavior, so its on by default

File: ext_libs/README (hunk 2 above)
Comment (jackgerrits) [reply ext_libs/README:1]: Seems reasonable

**Review by bassmang — APPROVED** [review]

**Pull request merged**